README.md

```diff
@@ -1,113 +1,323 @@
 Act
 ===
 
-This project is an effort by several groups working on formal methods for Ethereum smart contracts, aiming at creating a simple but effective language to write formal specification.
+This project is an effort by several groups working on formal methods for
+Ethereum smart contracts, aiming at creating a simple but effective language
+to write formal specification.
+It extends on the previous
+[Act](https://github.com/dapphub/klab/blob/master/acts.md) project.
 
 Some general features it hopes to achieve are:
 - Simple structure and logic.
-- Support writing properties on a high level (closer to Solidity/Vyper) as well as on the machine level.
-- Cross-tool communication: Different tools might be able to generate different knowledge and use the specification language to exchange information.
-- No attachment to a particular logic. If a tool wishes to support certain expressions, extensions can be written.
+- Support writing properties on a high level (closer to Solidity/Vyper) as
+  well as on the machine level.
+- Cross-tool communication: Different tools might be able to generate
+  different knowledge and use the specification language to exchange
+  information.
+- No attachment to a particular logic. If a tool wishes to support certain
+  expressions, extensions can be written.
 
 Everyone is encouraged to join/start discussions on issues and pull requests.
 
-Starting Point
+
+Language
+========
+
+Acts are based on behaviours from the specified smart contract. Take the
+following toy state machine written in Solidity as an example:
+
+```sol
+contract StateMachine {
+    uint x;
+  
+    function f() public {      
+        if (x == 0)
+            x = 1;             
+    }
+
+    function g() public {
+        if (x == 1)
+            x = 0;
+    }
+}
+```
+
+The initial state is `x = 0`, and `x` can toggle between 1 and 0 by calling
+functions `f` and `g`, respectively.
+
+A simple specification for the program above could be the following Act:
+
+```act
+behaviour init of StateMachine
+interface constructor()
+
+creates
+	uint256 x := 0
+
+invariants
```

Comment:
ooh, I like how you put the invariants in the `contructor`!

Reply:
This might be a better idea than having loose hanging `property/invariant` sections.

Reply:
Yea they were loose before but there was kind of this idea of a super section `contract`. Since this doesn't exist I wasn't sure where to put them, so just put it there. I think it works too.